pkg/common/scheduler/state/interface.go

```diff
@@ -0,0 +1,209 @@
+/*
+Copyright 2021 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package state
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const (
+	PodFitsResources                   = "PodFitsResources"
+	NoMaxResourceCount                 = "NoMaxResourceCount"
+	EvenPodSpread                      = "EvenPodSpread"
+	AvailabilityNodePriority           = "AvailabilityNodePriority"
+	AvailabilityZonePriority           = "AvailabilityZonePriority"
+	LowestOrdinalPriority              = "LowestOrdinalPriority"
+	RemoveWithEvenPodSpreadPriority    = "RemoveWithEvenPodSpreadPriority"
+	RemoveWithAvailabilityNodePriority = "RemoveWithAvailabilityNodePriority"
+	RemoveWithAvailabilityZonePriority = "RemoveWithAvailabilityZonePriority"
+	RemoveWithHighestOrdinalPriority   = "RemoveWithHighestOrdinalPriority"
+)
```

Comment:
this feels strange as these are plugins and this file should know nothing about them, right?

Reply:
@lionelvillard Interface.go turned out to be like the common place to store all the plugin name strings and arg details. Mainly for reuse (between similar scheduler and descheduler plugins) and also to expose info to the `Autoscaler`. Autoscaler HA mode is turned on (Scale up/down by number of domains) by checking the presence of an HA plugin like `AvailabilityZonePriority`